workspaces/marketplace/packages/cli/src/commands/export-csv/index.ts

```diff
@@ -0,0 +1,241 @@
+    /** The getter function for each cell. The return value must be castable to a string */
+    getter: (plugin: T) => any;
+  }[];
+
+  /** The rows to be included in the CSV */
+  private readonly rows: T[];
+
+  constructor(
+    columns: CSVGenerator<T>['columns'],
+    rows: CSVGenerator<T>['rows'] = [],
+  ) {
+    this.columns = columns;
+    this.rows = rows;
+  }
+
+  /** Escape problematic characters in a CSV cell */
+  static readonly escapeCSVCell = (cell: string) => {
+    const removedNewLines = cell.replace(/\n/g, '\\n');
+    const regex = /[\s,"]/;
+
+    // adapted from https://stackoverflow.com/a/47340986
+    if (regex.exec(removedNewLines.replace(/ /g, ''))) {
+      return `"${removedNewLines.replace(/"/g, '""')}"`;
+    }
+
+    return cell;
+  };
+
+  /** Add a row to the CSV */
+  addRow(row: T) {
+    this.rows.push(row);
+  }
+
+  /**
+   * @param includeHeader Whether to include the header row in the CSV
+   * @returns A CSV generated from the rows and columns provided
+   */
+  generate(includeHeader = true): string {
+    const header = this.columns.map(column => column.key).join(',');
+    const rows = this.rows
+      .map(row =>
+        this.columns
+          .map(column =>
+            CSVGenerator.escapeCSVCell(String(column.getter(row) || '')),
+          )
+          .join(','),
+      )
+      .join('\n');
+
+    return includeHeader ? `${header}\n${rows}` : rows;
+  }
+}
+
+/**
+ * Zips two arrays `keys` and `getters` into an array of CSVGenerator columns.
+ *
+ * `keys` and `getters` must have the same length.
+ *
+ * @param keys The keys to be used in the CSV header
+ * @param getters The getter functions for each cell
+ */
+const getColumns = <T = any>(keys: string[], getters: ((obj: T) => any)[]) =>
+  keys.map((key, index) => ({ key, getter: getters[index] }));
+
+/**
+ * A type guard for Entity
+ *
+ * @param entity The entity to check
+ * @returns Whether the entity is an Entity
+ */
+const isEntity = (entity: any): entity is Entity => {
+  return (
+    !!entity &&
+    typeof entity === 'object' &&
+    'apiVersion' in entity &&
+    'kind' in entity
+  );
+};
+
+/**
+ * Parse a list of YAML files into a list of entities
+ * @param yamls The list of paths of YAML files to parse
+ * @returns A list of entities
+ */
+const parseYamls = async (yamls: string[]) => {
+  return await Promise.all(
+    (yamls || []).flatMap(basePath => {
+      return fs
+        .readdirSync(basePath)
+        .filter(f => f.endsWith('.yaml'))
+        .map(async (file: string) => {
+          const fileContent = await fs.readFile(
+            path.join(basePath, file),
+            'utf-8',
+          );
+
+          // read each individual plugin from the file
+          const plugins = fileContent
+            .split('\n---\n')
+            .filter(Boolean)
+            .map((plugin: string) => {
+              return YAML.parse(plugin);
+            })
+            .filter(isEntity);
+          return plugins;
```

Comment:
Hey, the YAML library supports parsing multiple docs via `YAML.parseAllDocuments`.  Each document should have something like toJSON.

```suggestion
          const plugins = YAML.parseAllDocuments(fileContent)
            .map((doc => doc.toJSON())
            .filter(isMarketplacePlugin)
          return plugins;
```